src/Mvc/Mvc.Core/src/Builder/OpenApiEndpointConventionBuilderExtensions.cs

```diff
@@ -0,0 +1,114 @@
+            params string[] additionalContentTypes)
+        {
+            return Produces(builder, statusCode, typeof(TResponse), contentType, additionalContentTypes);
+        }
+
+        /// <summary>
+        /// Adds the <see cref="ProducesResponseTypeAttribute"/> to <see cref="EndpointBuilder.Metadata"/> for all builders
+        /// produced by <paramref name="builder"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="MinimalActionEndpointConventionBuilder"/>.</param>
+        /// <param name="statusCode">The response status code.</param>
+        /// <param name="responseType">The type of the response. Defaults to null.</param>
+        /// <param name="contentType">The response content type. Defaults to "application/json" if responseType is not null, otherwise defaults to null.</param>
+        /// <param name="additionalContentTypes">Additional response content types the endpoint produces for the supplied status code.</param>
+        /// <returns>A <see cref="MinimalActionEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+#pragma warning disable RS0026
+        public static MinimalActionEndpointConventionBuilder Produces(this MinimalActionEndpointConventionBuilder builder,
+#pragma warning restore RS0026
+            int statusCode,
+            Type? responseType = null,
+            string? contentType = null,
+            params string[] additionalContentTypes)
+        {
+            if (responseType is Type && string.IsNullOrEmpty(contentType))
+            {
+                contentType = "application/json";
+            }
+
+            if (contentType is null)
+            {
+                builder.WithMetadata(new ProducesResponseTypeAttribute(responseType ?? typeof(void), statusCode));
+                return builder;
+            }
+
+            builder.WithMetadata(new ProducesResponseTypeAttribute(responseType ?? typeof(void), statusCode, contentType, additionalContentTypes));
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Adds the <see cref="ProducesResponseTypeAttribute"/> with a <see cref="ProblemDetails"/> type
+        /// to <see cref="EndpointBuilder.Metadata"/> for all builders produced by <paramref name="builder"/>. 
+        /// </summary>
+        /// <param name="builder">The <see cref="MinimalActionEndpointConventionBuilder"/>.</param>
+        /// <param name="statusCode">The response status code.</param>
+        /// <param name="contentType">The response content type. Defaults to "application/problem+json".</param>
+        /// <returns>A <see cref="MinimalActionEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static MinimalActionEndpointConventionBuilder ProducesProblem(this MinimalActionEndpointConventionBuilder builder,
+            int statusCode,
+            string contentType = "application/problem+json")
```

Comment:
Could we make it consistent and have these be nullable like the other API?
```suggestion
            string? contentType = null)
```

Reply:
Same for the other one.